### PR TITLE
server: remove the useless and annoying error stacks in the log

### LINKED ISSUE #15554: UCP: remove error stack in log file when table doesn't exist
## Description

Executing a SQL which involves a not existed table, such as the following:

```sql
TiDB(root@127.0.0.1:test) > explain select * from t_not_exist;
ERROR 1146 (42S02): Table 'test.t_not_exist' doesn't exist
```

the log file looks like this:

```
[2020/03/22 12:09:15.413 +08:00] [WARN] [conn.go:700] ["command dispatched failed"] [conn=1] [connInfo="id:1, addr:127.0.0.1:63618 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="explain select * from t_not_exist"] [txn_mode=OPTIMISTIC] [err="[schema:1146]Table 'test.t_not_exist' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\t/opt/gopath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/parser/terror.(*Error).GenWithStackByArgs\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/terror/terror.go:243\ngithub.com/pingcap/tidb/infoschema.(*infoSchema).TableByName\n\t/Users/jianzhang.zj/Code/tidb/infoschema/infoschema.go:144\ngithub.com/pingcap/tidb/planner/core.(*preprocessor).handleTableName\n\t/Users/jianzhang.zj/Code/tidb/planner/core/preprocess.go:833\ngithub.com/pingcap/tidb/planner/core.(*preprocessor).Leave\n\t/Users/jianzhang.zj/Code/tidb/planner/core/preprocess.go:203\ngithub.com/pingcap/parser/ast.(*TableName).Accept\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:305\ngithub.com/pingcap/parser/ast.(*TableSource).Accept\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:442\ngithub.com/pingcap/parser/ast.(*Join).Accept\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:152\ngithub.com/pingcap/parser/ast.(*TableRefsClause).Accept\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:623\ngithub.com/pingcap/parser/ast.(*SelectStmt).Accept\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:966\ngithub.com/pingcap/parser/ast.(*ExplainStmt).Accept\n\t/opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/misc.go:241\ngithub.com/pingcap/tidb/planner/core.Preprocess\n\t/Users/jianzhang.zj/Code/tidb/planner/core/preprocess.go:56\ngithub.com/pingcap/tidb/executor.(*Compiler).Compile\n\t/Users/jianzhang.zj/Code/tidb/executor/compiler.go:57\ngithub.com/pingcap/tidb/session.(*session).execute\n\t/Users/jianzhang.zj/Code/tidb/session/session.go:1130\ngithub.com/pingcap/tidb/session.(*session).Execute\n\t/Users/jianzhang.zj/Code/tidb/session/session.go:1084\ngithub.com/pingcap/tidb/server.(*TiDBContext).Execute\n\t/Users/jianzhang.zj/Code/tidb/server/driver_tidb.go:248\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/Users/jianzhang.zj/Code/tidb/server/conn.go:1230\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/Users/jianzhang.zj/Code/tidb/server/conn.go:871\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/Users/jianzhang.zj/Code/tidb/server/conn.go:685\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/Users/jianzhang.zj/Code/tidb/server/server.go:416\nruntime.goexit\n\t/opt/goroot/src/runtime/asm_amd64.s:1357"]
```

The formatted error message is the following:

```
[2020/03/22 12:09:15.413 +08:00] [WARN] [conn.go:700] ["command dispatched failed"] [conn=1] [connInfo="id:1, addr:127.0.0.1:63618 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="explain select * from t_not_exist"] [txn_mode=OPTIMISTIC] [err="[schema:          1146]Table 'test.t_not_exist' doesn't exist
github.com/pingcap/errors.AddStack
    /opt/gopath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174
github.com/pingcap/parser/terror.(*Error).GenWithStackByArgs
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/terror/terror.go:243
github.com/pingcap/tidb/infoschema.(*infoSchema).TableByName
    /Users/jianzhang.zj/Code/tidb/infoschema/infoschema.go:144
github.com/pingcap/tidb/planner/core.(*preprocessor).handleTableName
    /Users/jianzhang.zj/Code/tidb/planner/core/preprocess.go:833
github.com/pingcap/tidb/planner/core.(*preprocessor).Leave
    /Users/jianzhang.zj/Code/tidb/planner/core/preprocess.go:203
github.com/pingcap/parser/ast.(*TableName).Accept
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:305
github.com/pingcap/parser/ast.(*TableSource).Accept
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:442
github.com/pingcap/parser/ast.(*Join).Accept
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:152
github.com/pingcap/parser/ast.(*TableRefsClause).Accept
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:623
github.com/pingcap/parser/ast.(*SelectStmt).Accept
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/dml.go:966
github.com/pingcap/parser/ast.(*ExplainStmt).Accept
    /opt/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200317021010-cd90cc2a7d87/ast/misc.go:241
github.com/pingcap/tidb/planner/core.Preprocess
    /Users/jianzhang.zj/Code/tidb/planner/core/preprocess.go:56
github.com/pingcap/tidb/executor.(*Compiler).Compile
    /Users/jianzhang.zj/Code/tidb/executor/compiler.go:57
github.com/pingcap/tidb/session.(*session).execute
    /Users/jianzhang.zj/Code/tidb/session/session.go:1130
github.com/pingcap/tidb/session.(*session).Execute
    /Users/jianzhang.zj/Code/tidb/session/session.go:1084
github.com/pingcap/tidb/server.(*TiDBContext).Execute
    /Users/jianzhang.zj/Code/tidb/server/driver_tidb.go:248
github.com/pingcap/tidb/server.(*clientConn).handleQuery
    /Users/jianzhang.zj/Code/tidb/server/conn.go:1230
github.com/pingcap/tidb/server.(*clientConn).dispatch
    /Users/jianzhang.zj/Code/tidb/server/conn.go:871
github.com/pingcap/tidb/server.(*clientConn).Run
    /Users/jianzhang.zj/Code/tidb/server/conn.go:685
github.com/pingcap/tidb/server.(*Server).onConn
    /Users/jianzhang.zj/Code/tidb/server/server.go:416
runtime.goexit
    /opt/goroot/src/runtime/asm_amd64.s:1357"]
```

This task is to remove the useless and annoying error stacks in the log, just need to print the error message.

## Score

* 50

## Mentor(s)

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

## Recommended Skills

* Golang Programming
* Golang Error Message

## Learning Materials

* [Golang Error](https://golang.org/pkg/errors/)
* Chinese: [TiDB 精选技术讲解文章](https://github.com/pingcap/presentations/blob/master/hackathon-2019/reference-document-of-hackathon-2019.md)
* English: [Awesome-Database-Learning](https://github.com/pingcap/awesome-database-learning)

### PR DESCRIPTION
**Pull request by wq1019**
Signed-off-by: wq1019 <2013855675@qq.com>

### What problem does this PR solve?

Issue Number: close #15554  

Problem Summary:
Executing a SQL which involves a not existed table, stack info will be printed in the log file.
So, The task is to remove the useless and annoying error stacks in the log.

### What is changed and how it works?
Delete code to print error stack in log.

What's Changed:
```
zap.String("err", errStrForLog(err)),	
```

Tests 

- Unit test

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -703,7 +702,6 @@ func (cc *clientConn) Run(ctx context.Context) {
 				zap.String("status", cc.SessionStatusToString()),
 				zap.Stringer("sql", getLastStmtInConn{cc}),
 				zap.String("txn_mode", txnMode),
-				zap.String("err", errStrForLog(err)),
```
Comment (zz-jason) [inline server/conn.go]: please do not remove the error message from the log.

File: server/conn.go [hunk 2]
```diff
@@ -737,14 +735,6 @@ func queryStrForLog(query string) string {
 	return query
 }
 
-func errStrForLog(err error) string {
-	if kv.ErrKeyExists.Equal(err) || parser.ErrParse.Equal(err) {
```
Comment (zz-jason) [inline server/conn.go]: instead, you need to add the error that doesn't need the stack info in here.

File: server/conn.go [hunk 3]
```diff
@@ -40,6 +40,8 @@ import (
 	"crypto/tls"
 	"encoding/binary"
 	"fmt"
+	"github.com/pingcap/tidb/infoschema"
```
Comment (zz-jason) [inline server/conn.go]: please group the third-part packages together.

**Comment by wq1019** [comment]
@jackysp  @sre-bot  Why is pr merged but there is no notice of score? :1234:

**Comment by zz-jason** [comment]
> @jackysp @sre-bot Why is pr merged but there is no notice of score? 🔢

I think this is because you haven't observe the [Format requirement](https://github.com/pingcap/tidb/pull/15607#issuecomment-603389756) for challenge program.

**Comment by zz-jason** [comment]
you can contact @you06 in #challenge-program channel for help.

**Pull request merged**